src/cam_qhy.cpp

```diff
@@ -556,6 +900,25 @@ bool Camera_QHY::Capture(int duration, usImage& img, int options, const wxRect&
             Debug.Write(wxString::Format("QHY set gain ret %d\n", (int) ret));
             pFrame->Alert(_("Failed to set camera gain"));
         }
+
+        if (m_offset < m_offsetMin)
+        {
+            m_offset = m_offsetMin;
+        }
+
+        if (m_offset > m_offsetMax)
+        {
+            m_offset = m_offsetMax;
+        }
+
+        double offset = floor(m_offset / m_offsetStep) * m_offsetStep;
```

Comment:
can we add some protection against `m_offsetStep == 0`, either here or back when we call `GetQHYCCDParamMinMaxStep`?